slack_bolt/context/ack/internals.py

```diff
@@ -1,42 +1,72 @@
-from typing import Optional, List, Union, Any
+from typing import Optional, List, Union, Any, Dict
 
 from slack_sdk.models.attachments import Attachment
 from slack_sdk.models.blocks import Block, Option, OptionGroup
+from slack_sdk.models.views import View
 
 from slack_bolt.error import BoltError
 from slack_bolt.response import BoltResponse
-from slack_bolt.util.utils import convert_to_dict_list
+from slack_bolt.util.utils import convert_to_dict_list, _to_dict
 
 
 def _set_response(
     self: Any,
     text_or_whole_response: Union[str, dict] = "",
     blocks: Optional[List[Union[dict, Block]]] = None,
     attachments: Optional[List[Union[dict, Attachment]]] = None,
+    response_type: Optional[str] = None,  # in_channel / ephemeral
+    # block_suggestion / dialog_suggestion
     options: Optional[List[Union[dict, Option]]] = None,
     option_groups: Optional[List[Union[dict, OptionGroup]]] = None,
+    # view_submission
+    response_action: Optional[str] = None,
+    errors: Optional[Dict[str, str]] = None,
+    view: Optional[Union[dict, View]] = None,
 ) -> BoltResponse:
     if isinstance(text_or_whole_response, str):
         text: str = text_or_whole_response
+        body = {"text": text}
+        if response_type:
+            body["response_type"] = response_type
         if attachments and len(attachments) > 0:
-            self.response = BoltResponse(
-                status=200,
-                body={"text": text, "attachments": convert_to_dict_list(attachments),},
+            body.update(
+                {"text": text, "attachments": convert_to_dict_list(attachments)}
             )
+            self.response = BoltResponse(status=200, body=body)
         elif blocks and len(blocks) > 0:
-            self.response = BoltResponse(
-                status=200, body={"text": text, "blocks": convert_to_dict_list(blocks),}
-            )
+            body.update({"text": text, "blocks": convert_to_dict_list(blocks)})
+            self.response = BoltResponse(status=200, body=body)
         elif options and len(options) > 0:
-            self.response = BoltResponse(
-                status=200, body={"options": convert_to_dict_list(options),}
-            )
+            body = {"options": convert_to_dict_list(options)}
+            self.response = BoltResponse(status=200, body=body)
         elif option_groups and len(option_groups) > 0:
-            self.response = BoltResponse(
-                status=200, body={"option_groups": convert_to_dict_list(option_groups),}
-            )
+            body = {"option_groups": convert_to_dict_list(option_groups)}
+            self.response = BoltResponse(status=200, body=body)
+        elif response_action:
+            # These patterns are in response to view_submission requests
```

Comment:
If I'm reading this correctly, `response_action == errors` is only ever associated with `view_submission` requests?

Reply:
>response_action == errors is only ever associated with view_submission requests?

Yes, it is (at least at this point)